UnityProject/Assets/Scripts/Messages/Server/TabUpdateMessage.cs

```diff
@@ -15,10 +18,60 @@ public class TabUpdateMessage : ServerMessage
 
 	private static readonly ElementValue[] NoValues = new ElementValue[0];
 
+	private static Dictionary<int, Tuple<ElementValue[], int>> ElementValuesCache = new Dictionary<int, Tuple<ElementValue[], int>>();
+
+	private static HashSet<int> ServerUniqueIdCache = new HashSet<int>();
+
+	public int ID;
```

Comment:
create a new `enum TabMessageType {MoreIncoming, EndOfMessage}` for that instead of calling it "ID" if it's just going to be "1" or "2"

Reply:
Done